packages/utils/src/useControlled.ts

```diff
@@ -74,3 +75,66 @@ export function useControlled<T = unknown>({
 
   return [value as T, setValueIfUncontrolled];
 }
+
+interface UseStoreControlledParameters<Value, State> {
+  controlled?: Value;
+  default?: Value;
+  store: Store<State>;
+  selector: (state: State) => Value;
+  name: keyof State;
+}
+
+export function useStoreControlled<Value, State>({
+  controlled,
+  default: defaultProp,
+  store,
+  selector,
+  name,
+}: UseStoreControlledParameters<Value, State>): [Value, (newValue: Value) => void] {
```

Comment:
Does this somehow bypass this issue: https://github.com/mui/base-ui/pull/2707#issuecomment-3272069030 

Or does it just avoid needing to use effects to synchronize the values to the store for the controlled props?